### PR TITLE
Add version() and gitCommitHash()

### PR DESCRIPTION
**Pull request by aprokop**

### CODE REVIEW

**Comment by aprokop** [comment]
The second tested could not build container for some reason, failed on LLVM step.

File: src/ArborX_Config.hpp.in [hunk 1]
```diff
@@ -12,6 +12,9 @@
 #ifndef ARBORX_CONFIG_HPP
 #define ARBORX_CONFIG_HPP
 
+#cmakedefine ArborX_VERSION_STRING "@ArborX_VERSION_STRING@"
+#cmakedefine ArborX_GIT_COMMIT_HASH "@ArborX_GIT_COMMIT_HASH@"
```
Comment (dalg24) [inline src/ArborX_Config.hpp.in]: All caps please (c.f. https://github.com/arborx/ArborX/pull/42#discussion_r281214386 :D)

Comment (aprokop) [reply src/ArborX_Config.hpp.in]: So, the funny thing, I tried all caps first. I had
```CMake
set(ArborX_VERSION_STRING "0.9 (dev)")
```
in `CMakeLists.txt`, and
```c++
#cmakedefine ARBORX_VERSION_STRING "@ArborX_VERSION_STRING@"
```
here. But... it did not define it in the resulting `ArborX_Config.hpp` (`#undef`). I have no clue why that is.

Comment (masterleinad) [reply src/ArborX_Config.hpp.in]: [`CMake`](https://cmake.org/cmake/help/v3.14/command/configure_file.html?highlight=cmakedefine) says:
> ```
> #cmakedefine VAR ...
> ```
> will be replaced with either:
> ```
> #define VAR ...
> ```
> or:
> ```
> /* #undef VAR */
> ```
> depending on whether `VAR` is set in `CMake` to any value not considered a false constant by the `if()` command.

Comment (aprokop) [reply src/ArborX_Config.hpp.in]: Thanks @masterleinad! I was too lazy to look it up, but it confirmed what I suspected: you can't have your cake and eat it too. I.e., I would have had to declare the variable as `ARBORX_VERSION_STRING` in CMake too, which I did not want to do :|

Comment (masterleinad) [reply src/ArborX_Config.hpp.in]: I tried to avoid that in #42, too. But then someone complained. :wink:

Comment (aprokop) [reply src/ArborX_Config.hpp.in]: Oh, yeah, I did not realize that was the exactly same case 🤦‍♂ . I apologize. I'll change it here too.

**Comment by aprokop** [comment]
@dalg24 Addressed your comments. This also uncovered a bug from copy-paste (`#ifndef` guard).

File: CMakeLists.txt [hunk 2]
```diff
@@ -33,6 +33,20 @@ install(EXPORT ArborXTargets
...
+    FIND_PACKAGE(Git)
+    IF(GIT_FOUND)
+        EXECUTE_PROCESS(
+            COMMAND           ${GIT_EXECUTABLE} log --pretty=format:%h -n 1
+            WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
+            OUTPUT_VARIABLE   ARBORX_GIT_COMMIT_HASH
+            )
+    ENDIF()
+ENDIF()
+MESSAGE("ArborX hash = '${ARBORX_GIT_COMMIT_HASH}'")
```
Comment (dalg24) [inline CMakeLists.txt]: Question:  Does this imply I need to reconfigure to update the hash?  (not complaining just wondering band making sure I understand)

File: src/ArborX_Version.hpp [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+#include <ArborX_Config.hpp>
+
+#include <string>
+
+namespace ArborX
+{
+
+inline std::string version() { return ARBORX_VERSION_STRING; }
+
+inline std::string gitCommitHash() { return ARBORX_GIT_COMMIT_HASH; }
```
Comment (dalg24) [inline src/ArborX_Version.hpp]: Do we really care about inlining for these?

File: CMakeLists.txt (hunk 2 above)
Comment (aprokop) [reply CMakeLists.txt]: I believe so.

File: src/ArborX_Version.hpp (hunk 3 above)
Comment (aprokop) [reply src/ArborX_Version.hpp]: I see no downside to it.

File: CMakeLists.txt [hunk 4]
```diff
@@ -33,6 +33,20 @@ install(EXPORT ArborXTargets
   DESTINATION lib/cmake/ArborX
 )
 
+SET(ARBORX_VERSION_STRING "0.9 (dev)")
```
Comment (dalg24) [inline CMakeLists.txt]: Will need to be sync'd up with versioning in CMake but good enough for now

**Pull request merged**